hydromt_wflow/wflow.py

```diff
@@ -1859,24 +1858,39 @@ def write_forcing(
                         self.write_config()  # re-write config
                         fn_out = fn_default_path
 
+            # Check if all dates between (starttime, endtime) are in all da forcing
+            start = pd.to_datetime(self.get_config("starttime"))
+            end = pd.to_datetime(self.get_config("endtime"))
+            missings = False
+            for da in self.forcing.values():
+                if "time" in da.coords:
+                    times = da.time.values
+                    if pd.to_datetime(start) < pd.to_datetime(times[0]):
```

Comment:
start is already a datetime object, you can skip the `pd.to_datetime(start)` here. And same for end in next if-statement. See also comment on L1875